documentation/src/main/asciidoc/user_guide/configuration.adoc

```diff
@@ -58,23 +54,96 @@ Cache defaultCache = manager.getCache();
 
-Refer to Infinispan configuration link:http://docs.jboss.org/infinispan/{infinispanversion}/configdocs[reference] for more details.
-If you are using XML editing tools for configuration writing you can use provided Infinispan link:http://infinispan.org/schemas/infinispan-config-{infinispanversion}.xsd[schema] to assist you. 
+The following example shows the simplest possible configuration for each of the cache types supported by Infinispan:
+
+[source,xml]
+----
+
+<infinispan>
+   <cache-container default-cache="local">
+      <transport cluster="mycluster"/>
+      <local-cache name="local"/>
+      <invalidation-cache name="invalidation" mode="SYNC"/>
+      <replicated-cache name="repl-sync" mode="SYNC"/>
+      <distributed-cache name="dist-sync" mode="SYNC"/>
+   </cache-container>
+</infinispan>
+
+----
+
+==== Cache configuration templates
+
+As mentioned above, Infinispan's supports the notion of _configuration templates_. These are full or partial configuration
+declarations which can be shared among multiple caches or as the basis for more complex configurations.
+
+The following example shows how a configuration named `local-template` is used to define a cache named `local`.
+
+[source,xml]
+----
+
+<infinispan>
+   <cache-container default-cache="local">
+      <!-- template configurations -->
+      <local-cache-configuration name="local-template">
+         <expiration interval="10000" lifespan="10" max-idle="10"/>
+      </local-cache-configuration>
+
+      <!-- cache definitions -->
+      <local-cache name="local" configuration="local-template" />
+   </cache-container>
+</infinispan>
+
+----
+
+Templates can inherit from previously defined templates, augmenting and/or overriding some or all of the configuration elements:
+
+[source,xml]
+----
+
+<infinispan>
+   <cache-container default-cache="local">
+      <!-- template configurations -->
+      <local-cache-configuration name="base-template">
+         <expiration interval="10000" lifespan="10" max-idle="10"/>
+      </local-cache-configuration>
+
+      <local-cache-configuration name="extended-template" configuration="base-template">
+         <expiration lifespan="20"/>
+         <eviction max-entries="2000" strategy="LRU"/>
+      </local-cache-configuration>
+
+      <!-- cache definitions -->
+      <local-cache name="local" configuration="base-template" />
+      <local-cache name="local-bounded" configuration="extended-template" />
+   </cache-container>
+</infinispan>
+
+----
+
+In the above example, `base-template` defines a local cache with a specific _expiration_ configuration. The `extended-template`
+configuration inherits from `base-template`, overriding just a single parameter of the _expiration_ element (all other
+attributes are inherited) and adds an _eviction_ element. Finally, two caches are defined: `local` which uses the `base-template`
+configuration and `local-bounded` which uses the `extended-template` configuration.
+
```

Comment:
I actually found a bug in this behaviour, so I've opened https://github.com/infinispan/infinispan/pull/4591